plugins/aws_sts/lib/samson_aws_sts/samson_plugin.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module SamsonAwsSts
+  class Engine < Rails::Engine
+  end
+
+  class << self
+    def deploy_env_vars(sts_client:, deploy:)
+      if deploy.stage.aws_sts_iam_role_arn.blank?
+        return {}
+      end
+
+      creds = sts_client.assume_role(
+        role_arn: deploy.stage.aws_sts_iam_role_arn,
+        role_session_name: session_name(deploy)
+      ).credentials
+
+      {
+        STS_AWS_ACCESS_KEY_ID:     "hidden://#{creds.access_key_id}",
+        STS_AWS_SECRET_ACCESS_KEY: "hidden://#{creds.secret_access_key}",
+        STS_AWS_SESSION_TOKEN:     "hidden://#{creds.session_token}"
+      }
+    end
+
+    def session_name(deploy)
+      "#{deploy.project.name}:#{deploy.stage.name}:deploy/#{deploy.id}"
```

Comment:
project and stage names can have all kinds of characters/lengths ... better check the docs on what a valid name is, then `.parameterize` and cut down to length if needed